File: SchoolMenu/SchoolMenu/ClientApp/src/app/models/Product.ts
export class Product {
    id: string;
    name: string;
    productPrice: number;

    constructor(id: string, name: string, productPrice: number) {
        this.id = id;
        this.name = name;
        this.productPrice = productPrice;
    }
}

export class AddProduct {
    name: string;
    productPrice: number;

    constructor(name: string, productPrice: number) {
        this.name = name;
        this.productPrice = productPrice;
    }
}

File: SchoolMenu/SchoolMenu.DAL/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SchoolMenu.DAL.Entities
{
    public class Product
    {
        [Key]
        public Guid Id { get; set; }
        public ICollection<Meal> Meals { get; set; }
        [MaxLength(128)]
        public string Name { get; set; }
        public decimal ProductPrice { get; set; }
    }
}
